vrecord

```diff
@@ -1184,67 +1194,74 @@ else
-        -f "${format}" "${dir}/${id}${suffix}.${extension}" \
-        "${extraoutputs[@]}" -map 0 -c copy -f nut - | \
+    else
         "${FFPLAY_DECKLINK[@]}" \
             -v info -hide_banner -stats -autoexit -i - \
             -window_title "mode:${runtype} - video:'${video_input_choice}' audio:'${audio_input_choice}'" \
             -vf "${playbackfilter}"
+    fi
+elif [[ "${media_player_choice}" = "mpv" ]] ; then
+    "${FFMPEG_DECKLINK[@]}" \
+    2> >(tee "${logdir}/${id}${capturelogsuffix}" /tmp/vrecord_input.log >/dev/null) \
+    -v info -nostdin -hide_banner -nostats "${inputoptions[@]}" "${GRAB_DECKLINK[@]}" \
+    -metadata:s:v:0 encoder="${codecname}" "${middleoptions[@]}" -c:a pcm_s24le \
+    -filter_complex "\
+        [0:v:0]${recordingfilter};\
+        ${audiomap}" \
+    $(if [ -n "$map1k" ] ; then echo ${map1k} ; fi) $(if [ -n "$map1v" ] ; then echo ${map1v} ; fi) \
+    $(if [ -n "$map2k" ] ; then echo ${map2k} ; fi) $(if [ -n "$map2v" ] ; then echo ${map2v} ; fi) \
+    -f "${format}" "${dir}/${id}${suffix}.${extension}" \
+    "${extraoutputs[@]}" -map 0 -c copy -f nut - | \
+    if [[ "${qctoolsxml_choice}" = "Yes" ]] ; then
+        tee >(mpv 2>/dev/null --scripts-append ./mpvs.lua -) | \
+            qcli -i - -o "${logdir}/${id}${suffix}.${extension}.qctools.xml.gz"
+    else
+        mpv 2>/dev/null --scripts-append ./mpvs.lua -
+    fi
 fi
 
-
-
 if [[ "${qctoolsxml_choice}" = "Yes" ]] ; then
+    if [[ ! "$(which qcli)" ]] ; then
+        _report -w "Please install qcli to use the qctools reporting option."
+        _report -w "Such as \`brew install qcli\`."
+        exit 1
+    fi
+    QCXML="${logdir}/${id}${suffix}.${extension}.qctools.xml.gz"
     _report -d "Vrecord is analyzing your video file. Please be patient."
     if [[ "${pixel_format}" = "yuv422p10" ]] ; then
-        SAT_OUTLIERS=$(gzcat "${logdir}/${id}${suffix}.${extension}.qctools.xml.gz" | xml sel -t -v "count(//tag[@key='lavfi.signalstats.SATMAX'][@value>496])" -n)
+        SAT_OUTLIERS=$(gzcat "${QCXML}" | xml sel -t -v "count(//tag[@key='lavfi.signalstats.SATMAX'][@value>496])" -n)
     elif [[ "${pixel_format}" = "uyvy422" ]] ; then
-        SAT_OUTLIERS=$(gzcat "${logdir}/${id}${suffix}.${extension}.qctools.xml.gz" | xml sel -t -v "count(//tag[@key='lavfi.signalstats.SATMAX'][@value>124])" -n)
+        SAT_OUTLIERS=$(gzcat "${QCXML}" | xml sel -t -v "count(//tag[@key='lavfi.signalstats.SATMAX'][@value>124])" -n)
+    fi
+    AUD_OUTLIERS=$(gzcat "${QCXML}" | xml sel -t -v "count(//tag[@key='lavfi.astats.Overall.Max_level'][@value>=0.999998])" -n)
+    BRNG_OUTLIERS=$(gzcat "${QCXML}" | xml sel -t -v "count(//tag[@key='lavfi.signalstats.BRNG'][@value>=0.03])" -n)
+    audio_peak=$(gzcat "${QCXML}" | grep lavfi.astats.Overall.Peak_level | cut -d '"' -f 4 | sort -n | tail -n 1)
+    _writeingestlog "Peak Volume is (dB)" "${audio_peak}"
+    if [[ ! -s "${XML}" ]] ; then
```

Comment:
Do you mean `${QCXML}`?

Reply:
also the above lines that use QCXML could be included in this condition to only run if the QCXML file is there, rather than attempt to use QCXML several times and then show a warning if it's not present later.